File: 136-Single Number.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # With extra memory:
        
        intkey = {}
        for i in nums:
            intkey[i] = intkey.get(i, 0) + 1
        
        for i, v in intkey.items(): 
            if intkey[i] == 1:
                return i

File: 136-Single Number.cpp
class Solution {
public:
    int singleNumber(vector<int>&nums){
        int res = 0;
        for(int i : nums)
            res ^= i;
        return res;
    }
    int singleNumber1(vector<int>& nums) {
        unordered_map<int, int> c;
        
        for(int i : nums){
            c[i] += 1;
        }
        
        for (auto &it : c){
            if(it.second == 1)
                return it.first;
        }
        return -1;
    }
};